### PR TITLE
fix WARNING: DATA RACE issue when multiple goroutines access the backend

### PR DESCRIPTION
**Pull request by ahrtr**
Fix [issues/13873](https://github.com/etcd-io/etcd/issues/13873).

One of the CI failures in [pull/13854](https://github.com/etcd-io/etcd/pull/13854) was also caused by this issue.

### CODE REVIEW

**Comment by ahrtr** [comment]
cc @ptabor @serathius @spzala

File: server/etcdserver/v3_server.go [hunk 1]
```diff
@@ -237,7 +237,11 @@ func (s *EtcdServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.
 		// the hash may revert to a hash prior to compaction completing
 		// if the compaction resumes. Force the finished compaction to
 		// commit so it won't resume following a crash.
+		//
+		// `applySnapshot` sets a new backend instance, so we need to acquire the bemu lock.
+		s.bemu.RLock()
 		s.be.ForceCommit()
```
Comment (ptabor) [inline server/etcdserver/v3_server.go:243]: It's interesting whether we have good test-coverage of 'swapping' backends in the middle of compaction.

The scheduledCompaction theoretically might finish on the 'old backends' why the recovered remains not compacted...

**Review by ptabor — APPROVED** [review]

**Comment by ptabor** [comment]
Thank you for fixing this.

File: server/etcdserver/v3_server.go (hunk 1 above)
Comment (ahrtr) [reply server/etcdserver/v3_server.go:243]: Let's discuss & fix the race condition between `compaction & defragmentation & snapshot` and `applySnapshot` cases in separate issues/PRs. I will drive it.

**Comment by ahrtr** [comment]
cc @serathius @spzala  Please also take a look. Thanks.

The fix in this PR can increase pipeline stability.

**Review by serathius — APPROVED** [review]

**Review by serathius — APPROVED** [review]

**Pull request merged**